[WIP] Docs for packaging as Docker image.

treeder opened the pull request:
For review.

treeder committed:
Docs for packaging as Docker image.

treeder committed:
Add private repository login.

treeder commented:
feedback requested @arschles @rdallman @rkononov

treeder committed:
Added step numbers.

treeder committed:
Explain payloads.

treeder committed:
Updated README.

arschles commented on ruby/README.md:
````diff
@@ -16,58 +16,109 @@ Notice in `hello.rb`, we add the following so it uses the vendored gems:
 
 ```sh
-docker build -t treeder/hello.rb:latest .
+iron docker login -e YOUR_DOCKER_HUB_EMAIL -p YOUR_DOCKER_HUB_PASSWORD
````
should leave a note here about how we store these credentials on our machines and what guarantees we make about their fidelity

rkononov commented:
lgtm, but need to add commands to ironcli before merging

rdallman commented:
ditto

treeder committed:
Added env vars on register.

treeder committed:
Updated with fix for Nokogiri build.

treeder committed:
Update README.md

Commit:
Update README.md

- missing "worker" to specify product
- also, the .rb for the docker image name is really confusing for someone going through this first time

Commit:
Update README.md

removing worker again

treeder committed:
Merge pull request #26 from iron-io/carimura-patch-1

Update README.md

treeder committed:
Merge branch 'image-packaging' of https://github.com/iron-io/dockerworker into image-packaging

treeder committed:
Added link to troubleshooting.

rkononov commented:
@treeder could we merge this one?

treeder commented:
Let me go through it and make sure everything works first.

treeder committed:
Updated README.

treeder committed:
Merge branch 'image-packaging' of https://github.com/iron-io/dockerworker into image-packaging

treeder commented:
Went through it all and updated a bit.  All works!  Should apply this to all languages.

rdallman commented:
lgtm
